bundles/statistics/statsgrid2016/view/SearchFlyout.js

```diff
@@ -85,12 +85,25 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.view.SearchFlyout', function (t
             }
 
             var newActiveIndicator = false;
-
+            
             selectedIndicators.forEach(function (indicator) {
+                var added;
                 if (indicator === '') {
                     return;
                 }
-                var added = me.service.getStateService().addIndicator(values.datasource, indicator, values.selections);
+                Object.keys(values.selections).forEach(function (key) {
+                    var selection = values.selections[key];
+                    if (Array.isArray(selection) && selection.length > 1) {
```

Comment:
Take out the selection.length > 1 as we will always want to reduce the array to single value.